src/Action/ExceptionAction.php

```diff
@@ -80,6 +83,12 @@ public function __invoke(FlattenException $exception, Request $request): Respons
         $headers['X-Content-Type-Options'] = 'nosniff';
         $headers['X-Frame-Options'] = 'deny';
 
-        return new Response($this->serializer->serialize($exception, $format['key']), $exception->getStatusCode(), $headers);
+        $response = new Response(null, $exception->getStatusCode(), $headers);
+
+        if (!$this->debug && $response->isServerError()) {
+            $exception->setMessage(Response::$statusTexts[$response->getStatusCode()]);
```

Comment:
@jordscream Have a look at https://github.com/symfony/symfony/blob/4d48b58d197cfa76046bb23ce9c2fe7bec8fbbb2/src/Symfony/Bundle/TwigBundle/Controller/ExceptionController.php#L41-L73

I see no compelling reason to modify the exception in any way whatsoever.

For example, https://github.com/api-platform/core/blob/8053c55311b713153179f412ea294362a9997bea/src/Action/ExceptionAction.php#L58-L84 can be rewritten as:

```php
/**
 * Converts a an exception to a JSON response.
 *
 * @param FlattenException $exception
 * @param Request          $request
 *
 * @return Response
 */
public function __invoke(FlattenException $exception, Request $request): Response
{
    $exceptionClass = $exception->getClass();
    $statusCode = $exception->getStatusCode();

    foreach ($this->exceptionToStatus as $class => $status) {
        if (is_a($exceptionClass, $class, true)) {
            $statusCode = $status;

            break;
        }
    }

    $headers = $exception->getHeaders();
    $format = ErrorFormatGuesser::guessErrorFormat($request, $this->errorFormats);
    $headers['Content-Type'] = sprintf('%s; charset=utf-8', $format['value'][0]);
    $headers['X-Content-Type-Options'] = 'nosniff';
    $headers['X-Frame-Options'] = 'deny';

    return new Response($this->serializer->serialize($exception, $format['key']), $statusCode, $headers);
}
```
with no change in behaviour and no BC.

> The Response with status code is dealed by the ExceptionAction, not by ErrorNormalizer....

The `ErrorNormalizer` is responsible for the body / content of the `Response`. (It basically replaces the content set when creating the `Response`.) How is it called? https://github.com/api-platform/core/blob/8053c55311b713153179f412ea294362a9997bea/src/EventListener/SerializeListener.php

So in our case, if we want to change the message, we should change in in the respective `ErrorNormalizer`s.